functions/src/delete-marked-readings.ts

```diff
@@ -0,0 +1,86 @@
+import {firestore} from './admin';
+import {readingsSubcollection} from './aqi-calculation/util';
+
+/**
+ * Deletes readings before date marked for sensors in the deletion list
+ */
+async function deleteMarkedReadings(): Promise<void> {
+  const batchSize = 2000;
+
+  const deletionMap =
+    (await firestore.collection('deletion').doc('todo').get()).data()
+      ?.deletionMap ?? Object.create(null);
+
+  for (const sensorDocId in deletionMap) {
+    const deleteBeforeDate = deletionMap[sensorDocId];
+    const readingsCollectionRef = firestore.collection(
+      readingsSubcollection(sensorDocId)
+    );
+    const query = readingsCollectionRef
+      .where('timestamp', '<', deleteBeforeDate)
+      .limit(batchSize);
+
+    deleteSensorSubcollectionBatch(query, batchSize, sensorDocId);
+  }
+}
+
+/**
+ * Wrapper function for `deleteQueryBatch` that uses a custom resolve function that removes the sensor from the deletion map
```

Comment:
Add comment that says this is only called if the query has size 0, meaning the deletion task is complete